src/main/kotlin/org/rust/ide/refactoring/move/RsMoveTopLevelItemsHandler.kt

```diff
@@ -58,19 +62,44 @@ class RsMoveTopLevelItemsHandler : MoveHandlerDelegate() {
         // otherwise we assume that user wants to move item under cursor
         val hasSelection = editor?.selectionModel?.hasSelection() ?: false
         if (hasSelection || canMove(elements, null, reference)) {
-            doMove(project, elements)
+            doMove(project, elements, editor)
             return true
         }
         return false
     }
 
-    private fun doMove(project: Project, elements: Array<out PsiElement>) {
-        val containingMod = PsiTreeUtil.findCommonParent(*elements)?.parentOfType<RsMod>() ?: return
-        val itemsToMove = elements.filterIsInstance<RsItemElement>().toSet()
+    private fun doMove(project: Project, elements: Array<out PsiElement>, editor: Editor?) {
+        val (itemsToMove, containingMod) = editor?.let { collectSelectedItems(project, editor) }
+            ?: run {
+                val containingMod = PsiTreeUtil.findCommonParent(*elements)?.parentOfType<RsMod>() ?: return
+                val items = elements.filterIsInstance<RsItemElement>().toList()
+                items to containingMod
+            }
 
         if (!CommonRefactoringUtil.checkReadOnlyStatusRecursively(project, itemsToMove, true)) return
 
-        RsMoveTopLevelItemsDialog(project, itemsToMove, containingMod).show()
+        RsMoveTopLevelItemsDialog(project, itemsToMove.toSet(), containingMod).show()
+    }
+
+    private fun collectSelectedItems(project: Project, editor: Editor): Pair<List<RsItemElement>, RsMod>? {
+        val selection = editor.selectionModel
+        val file = PsiDocumentManager.getInstance(project).getPsiFile(editor.document) ?: return null
```

Comment:
[minor] can be converted to `editor.document.toPsiFile(project)` using `org.rust.openapiext.UtilsKt#toPsiFile(com.intellij.openapi.editor.Document, com.intellij.openapi.project.Project)`